internal/webhooks/cluster.go

```diff
@@ -524,7 +526,29 @@ func (webhook *Cluster) pollClusterClassForCluster(ctx context.Context, cluster
 		if clusterClassGetErr = webhook.Client.Get(ctx, client.ObjectKey{Namespace: cluster.Namespace, Name: cluster.Spec.Topology.Class}, clusterClass); clusterClassGetErr != nil {
 			return false, nil //nolint:nilerr
 		}
+		// Return an error if the ClusterClass has not successfully reconciled either because variables aren't correctly
+		// reconciled or because the observed generation doesn't match the metadata generation.
+		if !conditions.Has(clusterClass, clusterv1.ClusterClassVariablesReconciledCondition) ||
+			conditions.IsFalse(clusterClass, clusterv1.ClusterClassVariablesReconciledCondition) ||
+			clusterClass.GetGeneration() != clusterClass.Status.ObservedGeneration {
+			clusterClassGetErr = errors.New("ClusterClass is not up to date. If this persists check ClusterClass status")
+			return false, nil
+		}
 		return true, nil
 	})
 	return clusterClass, clusterClassGetErr
 }
+
+func clusterClassStatusVariablesToVariables(vars []clusterv1.ClusterClassStatusVariable) []clusterv1.ClusterClassVariable {
+	var ccVars []clusterv1.ClusterClassVariable
+	for _, v := range vars {
```

Comment:
Q: This is sort of safe as an intermediate step as the ClusterClass reconciler doesn't allow multiple definitions per variable right now, right?

Reply:
Hm looks like it isn't as we can have inline and external patches with the same name right now

Reply:
Sorry just having problem keeping track of all the changes we did and the ones we will make in the future and then see if the changes we are doing are safe.

(I'm getting the feeling I have to re-review all changes in the last PR again to get the full picture)

Reply:
Yeah - this is updated in #8107 - but I think the intermediate version here is alright. This has the caveat that right now using external + inline variables is essentially undefined behaviour as the feature is only partially implemented. 

I'm not sure if this is something we need to ameliorate - if so we could take out the discover variables call and add it into the final part of the basic implementation - #8026

Reply:
As discussed - current state is fine for me